### ISSUE TITLE
Dependency Conflict with TensorFlow.js (pyyaml)

### ISSUE THREAD

**Issue by brycecoldcocacola** [tensorflow/models#11077]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/official

## 2. Describe the bug

TF-Vision cannot be installed alongside TensorFlow.js due to dependency issues with pyyaml.

## 3. Steps to reproduce

Within a [Poetry](https://python-poetry.org/) environment:

```
>>> poetry add tf-models-official
Using version ^2.13.1 for tf-models-official

Updating dependencies
Resolving dependencies... (105.7s)

Writing lock file

Package operations: 62 installs, 0 updates, 0 removals

  • Installing pyasn1 (0.5.0)
  • Installing cachetools (5.3.1)
  • Installing oauthlib (3.2.2)
  • Installing pyasn1-modules (0.3.0)
  • Installing rsa (4.9)
  • Installing google-auth (2.22.0)
  • Installing requests-oauthlib (1.3.1)
  • Installing absl-py (1.4.0)
  • Installing google-auth-oauthlib (1.0.0)
  • Installing grpcio (1.57.0)
  • Installing importlib-resources (6.0.1)
  • Installing markdown (3.4.4)
  • Installing protobuf (4.24.1)
  • Installing tensorboard-data-server (0.7.1)
  • Installing astunparse (1.6.3)
  • Installing etils (1.4.1)
  • Installing flatbuffers (23.5.26)
  • Installing gast (0.4.0)
  • Installing google-pasta (0.2.0)
  • Installing googleapis-common-protos (1.60.0)
  • Installing h5py (3.9.0)
  • Installing httplib2 (0.22.0)
  • Installing keras (2.13.1)
  • Installing libclang (16.0.6)
  • Installing opt-einsum (3.3.0)
  • Installing tensorboard (2.13.0)
  • Installing tensorflow-estimator (2.13.0)
  • Installing tensorflow-io-gcs-filesystem (0.33.0)
  • Installing termcolor (2.3.0)
  • Installing text-unidecode (1.3)
  • Installing wrapt (1.15.0)
  • Installing array-record (0.4.1)
  • Installing dm-tree (0.1.8)
  • Installing google-api-core (2.11.1)
  • Installing google-auth-httplib2 (0.1.0)
  • Installing lxml (4.9.3)
  • Installing portalocker (2.7.0)
  • Installing promise (2.3)
  • Installing python-slugify (8.0.1)
  • Installing regex (2023.8.8)
  • Installing tabulate (0.9.0)
  • Installing tensorflow (2.13.0)
  • Installing tensorflow-hub (0.14.0)
  • Installing tensorflow-metadata (1.13.1)
  • Installing uritemplate (4.1.1)
  • Installing cython (3.0.0)
  • Installing gin-config (0.5.0)
  • Installing google-api-python-client (2.97.0)
  • Installing immutabledict (3.0.0)
  • Installing kaggle (1.5.16)
  • Installing oauth2client (4.1.3)
  • Installing opencv-python-headless (4.8.0.76)
  • Installing py-cpuinfo (9.0.0)
  • Installing pycocotools (2.0.7)
  • Installing sacrebleu (2.3.1)
  • Installing sentencepiece (0.1.99)
  • Installing seqeval (1.2.2)
  • Installing tensorflow-datasets (4.9.2)
  • Installing tensorflow-model-optimization (0.7.5)
  • Installing tensorflow-text (2.13.0)
  • Installing tf-slim (1.1.0)
  • Installing tf-models-official (2.13.1)
>>> poetry add tensorflowjs
Using version ^4.10.0 for tensorflowjs

Updating dependencies
Resolving dependencies... (0.5s)

Because no versions of tensorflowjs match >4.10.0,<5.0.0
 and tensorflowjs (4.10.0) depends on flax (>=0.6.2,<0.6.3), tensorflowjs (>=4.10.0,<5.0.0) requires flax (>=0.6.2,<0.6.3).
And because no versions of flax match >0.6.2,<0.6.3
 and flax (0.6.2) depends on PyYAML (>=5.4.1), tensorflowjs (>=4.10.0,<5.0.0) requires PyYAML (>=5.4.1).
And because tf-models-official (2.13.1) depends on pyyaml (>=5.1,<5.4.0)
 and no versions of tf-models-official match >2.13.1,<3.0.0, tensorflowjs (>=4.10.0,<5.0.0) is incompatible with tf-models-official (>=2.13.1,<3.0.0).
So, because mlops depends on both tf-models-official (^2.13.1) and tensorflowjs (^4.10.0), version solving failed.
```

Given `tf-models-official` depends on PyYAML (>=5.1,<5.4.0) and `tensorflowjs` depends on PyYAML (>=5.4.1), they cannot be installed together.

Attempting to install earlier versions of `tensorflowjs` is not an option. This issue presents itself up to `tensorflowjs==4.5.0`, and then anything before that raises a dependency conflict with protobuf.

## 4. Expected behavior

Installing both `tf-models-official` and `tensorflowjs` within the same environment should be supported. It is reasonable for workflows where one wants to build a vision ML model and export it to a tfjs model format.

## 5. Additional context

N/A

## 6. System information

- OS Platform and Distribution: macOS 13.4.1 (c)
- TensorFlow installed from (source or binary): PyPi
- TensorFlow version (use command below): 2.13.0
- Python version: =3.9,<3.11

**Comment by joereger** [comment]
same issue, struggled with it last night.  is there any way to override the pyyaml version dependency and force an install?

**Comment by laxmareddyp** [comment]
Hi @brycecoldcocacola ,

We understand the problem you are having and we are working on a fix. The fix will be included in the next release of tf-models-official 2.14.0. Hope you understands.

Thanks.

**Comment by 2Andrey2** [comment]
Hello, maybe someone found a workaround? Or is the project currently inoperable? I have the same problem I can't install object_detection because of PyYAML.

**Comment by laxmareddyp** [comment]
Hi @brycecoldcocacola,

Our pull request (PR), which can be found [here](https://github.com/tensorflow/models/commit/623a471131e905436c3b21205075b031e1bc4795), has been successfully merged into the master branch of the TensorFlow models repository. This update will be included in the forthcoming 2.14.0 release.

Thanks.

**Comment by laxmareddyp** [comment]
Hi @brycecoldcocacola,

We are pleased to announce the release of version 2.13.2 of the tf-models-official package, now available on [PyPI](https://pypi.org/project/tf-models-official/). This release incorporates the aforementioned changes and improvements, and we can confirm that it is functioning as intended.

With this successful release, we consider the reported issue resolved and will proceed to close it. Thank you for your attention and support.
